datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -268,8 +271,7 @@ public void start() {
     // Initializing executor services
     _scheduledExecutor = Executors.newSingleThreadScheduledExecutor(
         new ThreadFactoryBuilder().setNameFormat("CoordinatorScheduledExecutor-%d").build());
-    // TODO Assess whether having a single threaded executor for token claim tasks is sufficient or it will be exhausted
-    _tokenClaimExecutor = Executors.newSingleThreadExecutor(
+    _tokenClaimExecutor = Executors.newFixedThreadPool(TOKEN_CLAIM_THREAD_POOL_SIZE,
```

Comment:
I think it's debatable what should be the ideal value for the pool size here. 

Only in the worst case when 8 concurrent stop requests occupy the thread pool of the leader and tokens assigned to itself are never scheduled for claiming, resulting in a deadlock situation. But this sounds very rare unless we do scripted restarts of datastreams in clusters.

Reply:
It's not even a deadlock situation. What happens is the executor gets backed up, and the task to claim tokens has to wait for threads to be available in order to be executed. So all tasks eventually get executed.
In case of a single threaded executor the token claim task will be executed after the poll task times out. So by the time it tries to claim the tokens leader already revokes them and declares a stop failure.

Reply:
As for the thread pool side, I'd rather be on the safer side, that's why I put 8.